homeassistant/helpers/debounce.py

```diff
@@ -104,3 +104,7 @@ def _schedule_timer(self) -> None:
             self.cooldown,
             lambda: self.hass.async_create_task(self._handle_timer_finish()),
         )
+        self.hass.async_create_task(self._wait_until_cooldown())
```

Comment:
I don't see what value this adds, you're just scheduling a task and no one is waiting for it ?

Reply:
Yeah, basically, and is not pretty.

This is what I was trying to tell you in the debouncer PR: tests rely on `block_til_done` to wait for all pendent tasks to finish, but debouncers set to immediate=False won't be awaited, as there are no tasks registered.

t=0 -> A new sample is received, debouncer starts cooldown (but no call to update)
t=0.001 -> Block til done is called to wait for the sample to be processed, but as there are no tasks, it returns instantly.
t=0.05 -> Cool down finishes and update is finally called (but it is too late for the test)

So I did that **for the statistics sensor tests to pass**, but I don't like it. I suppose I could insert artificial waits inside the tests, to make them pass, but I thought that option would be even uglier, not sure now. 

I was expecting to discuss how to approach this problem :) (which is only for tests, as in real-time debouncers end ok, and block_til_done() is not called)

Reply:
> I could insert artificial waits inside the tests, to make them pass, but I thought that option would be even uglier, not sure now.

**I reverted the change in debouncer and inserted the wait in the tests**, but the problem I see is that anyone using `hass.block_till_done()`, without specific knowledge of how the debouncer works, is going to expect that the call is made, so it could be a little confusing, right?

Reply:
We should not sleep in tests. That is slowing down our tests. Instead catch the call to schedule it, and run that directly.

Reply:
Ok, I'll make the change